comp/process/runner/runner.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/process/submitter"
+)
+
+// runner implements the Component.
+type runner struct {
+	checks    []Check
+	submitter submitter.Component
+}
+
+type dependencies struct {
+	fx.In
+
+	Checks    []Check `group:"check"`
+	Submitter submitter.Component
+}
+
+func newRunner(deps dependencies) (Component, error) {
+	return &runner{
+		checks:    deps.Checks,
+		submitter: deps.Submitter,
+	}, nil
+}
+
+func (r *runner) Run(exit <-chan struct{}) error {
+
+	for _, c := range r.checks {
+		if !c.IsEnabled() {
```

Comment:
Just keep in mind that disabled checks (and their dependencies) will be instantiated by Fx. Let me know if it is an issue then I could provide ideas solve this problem.

Reply:
Yes, this is something we discussed a couple of weeks ago, I am ok with it for now, we can decide on separate bundles later.